src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -203,31 +205,87 @@ public function getDefaultPageLength()
      */
     public function addCustomPageLengthToPageLengthMenu()
     {
-        // If the default Page Length isn't in the menu's values, Add it the beginnin and resort all to show a croissant list.
-        // assume both arrays are the same length.
-        if (! in_array($this->getDefaultPageLength(), $this->getOperationSetting('pageLengthMenu')[0])) {
-            // Loop through 2 arrays of prop. pageLengthMenu
-            foreach ($this->getOperationSetting('pageLengthMenu') as $key => &$page_length_choices) {
-                // This is a condition that should be always true.
-                if (is_array($page_length_choices)) {
-                    array_unshift($page_length_choices, $this->getDefaultPageLength());
+        $values = $this->getOperationSetting('pageLengthMenu')[0];
+        $labels = $this->getOperationSetting('pageLengthMenu')[1];
+        // this is a condition that should be always true.
+        if (is_array($values) && is_array($labels)) {
```

Comment:
```suggestion
        if (array_search($this->getDefaultPageLength(), $values) === false) {
```
@pxpm this is missing on the new setup. We need to validate the value doesn't exist.

Another thing, I removed here the validations to check if both values are array, for my observation, this values come here as being always arrays. Correct me if I'm wrong.

Reply:
Hey @promatik are you sure that we need to validate that the value doesn't exist ? 

It works in tests without it when adding a value that does not exist: 

```php
private $defaultPaginator = [[10, 20, 30], ['t1', 't2', 't3']];

    /**
     * Tests if table paginator adds default option non-existent at time in the paginator.~.
     */
    public function testCrudPanelPaginatorAddsDefaultOptionNonExistent()
    {
        $this->crudPanel->setModel(User::class);
        $this->crudPanel->setOperation('list');
        $this->crudPanel->setDefaultPageLength(40);
        $this->crudPanel->setPageLengthMenu($this->defaultPaginator);

        $this->assertCount(2, $this->crudPanel->getPageLengthMenu());
        $this->assertTrue(in_array(40, $this->crudPanel->getOperationSetting('pageLengthMenu')[0]));
        $this->assertEquals(array_values($this->crudPanel->getPageLengthMenu()[0])[3], 40);
    }
```

Technically it should be always array. I just wouldn't mind to have that extra check there.

Reply:
Just ignore me. 

You are right, I was talking about wrong test. Sorry.

Fixed it. Thanks again \o